### ISSUE TITLE
Menu bar

### ISSUE THREAD

**Issue by alexjpwalker** [typedb/typedb-studio#462]
Once we have all core features, we should add all user actions to a main menu. This would currently include:

- Open Project
- Connect to Database
- Save Current File
- Open Database Manager
- Close Transaction
- Commit Transaction
- Rollback Transaction
- Run Query
- Stop Signal

... and anything else we can think of.

And they should have the appropriate keyboard shortcuts.

**Comment by haikalpribadi** [comment]
This main menu is what meant by the "application menu bar" that appears on the OS top menu bar (for MacOS), right, @alexjpwalker @jamesreprise ?

**Comment by haikalpribadi** [comment]
Like, this would be the application menu bar for, Intellij on MacOS:

<img width="862" alt="Screenshot 2022-08-16 at 16 38 12" src="https://user-images.githubusercontent.com/567679/184920821-8258ef21-d70e-4349-ad2d-4c30f27da1a1.png">

This is what we're thinking, right?

**Comment by haikalpribadi** [comment]
If so, then perhaps we should also design what the application menu bar grouping (e.g. File, Edit, View, etc.) should be for TypeDB Studio.

**Comment by jamesreprise** [comment]
> This main menu is what meant by the "application menu bar" that appears on the OS top menu bar (for MacOS), right, @alexjpwalker @jamesreprise ?

That's correct. We discussed that the terminology 'Main Menu' was something that was familiar to @alexjpwalker from development for Windows but less familiar to me.

**Comment by haikalpribadi** [comment]
In that case issue #568 is sub-task of this issue, right?

**Comment by jamesreprise** [comment]
Right, this has already been flagged (see https://github.com/vaticle/typedb-studio/issues/568#issuecomment-1153765833).

**Comment by alexjpwalker** [comment]
After the app's redesign for Studio 3 I am no longer persuaded that a menu bar is necessary in any version of Studio 3.
